### PR TITLE
en, zh: optimize upgrade-tidb-operator

### PR DESCRIPTION
**Pull request by ran-huang**
Signed-off-by: Ran <huangran@pingcap.com>

### What is changed, added, or deleted? (Required)

Optimize upgrade-tidb-operator:
- Simplify steps.
- Refine wording.
- Remove the notice for v1.1

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.2 (TiDB Operator 1.2 versions)
- [ ] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s): 

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: zh/upgrade-tidb-operator.md [hunk 1]
```diff
@@ -6,24 +6,35 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/upgrade-tidb-operator/']
...
+本文介绍如何升级 TiDB Operator 到指定版本。你可以选择[在线升级](#在线升级)或[离线升级](#离线升级)。
 
- > **注意：**
- >
- > 你可以通过 `helm search repo -l tidb-operator` 命令查看当前支持的 TiDB Operator 版本。
- > 如果此命令的输出中未包含最新版本，可以使用 `helm repo update` 命令更新 repo。详情请参考[配置 Helm repo](tidb-toolkit.md#配置-helm-repo) 。
+## 前置检查
 
-## 在线升级步骤
+升级 TiDB Operator 前，确保当前 Helm 支持你需要升级的 TiDB Operator 版本。通过以下命令查看 Helm 支持的 TiDB Operator 版本：
```
Comment (DanielZhangQD) [inline zh/upgrade-tidb-operator.md:13]:
```suggestion
升级 TiDB Operator 前，确保 Helm repo 包含你需要升级的 TiDB Operator 版本。通过以下命令查看 Helm repo 包含的 TiDB Operator 版本：
```

Comment (DanielZhangQD) [reply zh/upgrade-tidb-operator.md:13]: 这是在线升级的一步，离线升级不需要

File: zh/upgrade-tidb-operator.md [hunk 2]
```diff
@@ -6,24 +6,35 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/upgrade-tidb-operator/']
...
 
     {{< copyable "shell-regular" >}}
 
     ```shell
-    kubectl apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/v1.2.4/manifests/crd.yaml && \
+    kubectl apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/${operatorImage}/manifests/crd.yaml && \
     kubectl get crd tidbclusters.pingcap.com
     ```
 
+    本文以 TiDB Operator v1.2.4 为例，你需要替换 `${operatorImage}` 为你要升级到的 TiDB Operator 版本。
```
Comment (DanielZhangQD) [inline zh/upgrade-tidb-operator.md:36]:
```suggestion
    本文以 TiDB Operator v1.2.4 为例，你需要替换 `${operatorImage}` 为你要升级到的 TiDB Operator 版本。
```
```suggestion
    本文以 TiDB Operator v1.2.4 为例，你需要替换 `${operatorVersion}` 为你要升级到的 TiDB Operator 版本。
```

File: zh/upgrade-tidb-operator.md [hunk 3]
```diff
@@ -60,30 +75,32 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/upgrade-tidb-operator/']
...
-   
+
         {{< copyable "shell-regular" >}}
 
         ```shell
-        wget https://raw.githubusercontent.com/pingcap/tidb-operator/v1.2.4/manifests/crd.yaml
+        wget https://raw.githubusercontent.com/pingcap/tidb-operator/${operatorImage}/manifests/crd.yaml
         ```
 
+        本文以 TiDB Operator v1.2.4 为例，你需要替换 `${operatorImage}` 为你要升级到的 TiDB Operator 版本。
```
Comment (DanielZhangQD) [inline zh/upgrade-tidb-operator.md:94]:
```suggestion
        本文以 TiDB Operator v1.2.4 为例，你需要替换 `${operatorVersion}` 为你要升级到的 TiDB Operator 版本。
```

**Comment by ran-huang** [comment]
@DanielZhangQD PTAL.

File: zh/upgrade-tidb-operator.md [hunk 4]
```diff
@@ -6,25 +6,34 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/upgrade-tidb-operator/']
...
+
+    如果输出中未包含你需要的新版本，可以使用 `helm repo update` 命令更新 repo。详情请参考[配置 Helm repo](tidb-toolkit.md#配置-helm-repo)。
+
+2. 更新 Kubernetes 的 CustomResourceDefinition (CRD)。关于 CRD 的更多信息，请参阅 [CustomResourceDefinition](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/)。
 
     {{< copyable "shell-regular" >}}
 
     ```shell
-    kubectl apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/v1.2.4/manifests/crd.yaml && \
+    kubectl apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/${operatorVersion}/manifests/crd.yaml && \
```
Comment (DanielZhangQD) [inline zh/upgrade-tidb-operator.md:30]:
```suggestion
    kubectl apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/${operator_version}/manifests/crd.yaml && \
```

File: zh/upgrade-tidb-operator.md [hunk 5]
```diff
@@ -6,25 +6,34 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/upgrade-tidb-operator/']
...
     kubectl get crd tidbclusters.pingcap.com
     ```
 
-2. 获取你要升级的 `tidb-operator` chart 中的 `values.yaml` 文件：
+    本文以 TiDB Operator v1.2.4 为例，你需要替换 `${operatorVersion}` 为你要升级到的 TiDB Operator 版本。
```
Comment (DanielZhangQD) [inline zh/upgrade-tidb-operator.md:34]:
```suggestion
    本文以 TiDB Operator v1.2.4 为例，你需要替换 `${operator_version}` 为你要升级到的 TiDB Operator 版本。
```

File: zh/upgrade-tidb-operator.md [hunk 6]
```diff
@@ -60,30 +73,32 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/upgrade-tidb-operator/']
...
+1. 使用有外网的机器，下载升级所需的文件和镜像。
 
     1. 下载 TiDB Operator 需要的 `crd.yaml` 文件。关于 CRD 的更多信息，请参阅 [CustomResourceDefinition](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/)。
-   
+
         {{< copyable "shell-regular" >}}
 
         ```shell
-        wget https://raw.githubusercontent.com/pingcap/tidb-operator/v1.2.4/manifests/crd.yaml
+        wget https://raw.githubusercontent.com/pingcap/tidb-operator/${operatorVersion}/manifests/crd.yaml
```
Comment (DanielZhangQD) [inline zh/upgrade-tidb-operator.md:89]:
```suggestion
        wget https://raw.githubusercontent.com/pingcap/tidb-operator/${operator_version}/manifests/crd.yaml
```

File: zh/upgrade-tidb-operator.md [hunk 7]
```diff
@@ -60,30 +73,32 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/upgrade-tidb-operator/']
...
         ```
 
+        本文以 TiDB Operator v1.2.4 为例，你需要替换 `${operatorVersion}` 为你要升级到的 TiDB Operator 版本。
```
Comment (DanielZhangQD) [inline zh/upgrade-tidb-operator.md:92]:
```suggestion
        本文以 TiDB Operator v1.2.4 为例，你需要替换 `${operator_version}` 为你要升级到的 TiDB Operator 版本。
```

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by lilin90 — APPROVED** [review]
LGTM

**Comment by ran-huang** [comment]
@lilin90 @DanielZhangQD PTAL at the English file. Thanks~

**Review by DanielZhangQD — APPROVED** [review]
LGTM

File: en/upgrade-tidb-operator.md [hunk 8]
```diff
@@ -71,98 +82,104 @@ This document describes how to upgrade TiDB Operator and Kubernetes.
 
     > **Note:**
     >
-    > After TiDB Operator is upgraded, the `discovery` deployment in all TiDB clusters will be automatically upgraded to the corresponding version of TiDB Operator.
+    > After TiDB Operator is upgraded, the `discovery` Deployment in all TiDB clusters is automatically upgraded to the corresponding version of TiDB Operator.
```
Comment (lilin90) [inline en/upgrade-tidb-operator.md:85]: @DanielZhangQD Is it necessary to capitalize "Deployment"? If no special reasons, I would suggest keeping it as an ordinary word without capitalization.

File: en/upgrade-tidb-operator.md [hunk 9]
```diff
@@ -171,10 +188,6 @@ If your server cannot access the Internet, you can take the following steps to u
...
-   > **Note:**
-   >
-   > After TiDB Operator is upgraded, the `discovery` deployment in all TiDB clusters will be automatically upgraded to the specified version of TiDB Operator.
-
-## Upgrade TiDB Operator from v1.0 to v1.1 or later releases
-
-Since TiDB Operator v1.1.0, PingCAP no longer updates or maintains the tidb-cluster chart. The components and features that have been managed using the tidb-cluster chart will be managed by CR (Custom Resource) or dedicated charts in v1.1. For more details, refer to [TiDB Operator v1.1 Notes](notes-tidb-operator-v1.1.md).
+    > **Note:**
+    >
+    > After TiDB Operator is upgraded, the `discovery` Deployment in all TiDB clusters is automatically upgraded to the corresponding version of TiDB Operator.
```
Comment (lilin90) [inline en/upgrade-tidb-operator.md:193]: Ditto.

**Review by lilin90 — APPROVED** [review]
LGTM

File: en/upgrade-tidb-operator.md (hunk 8 above)
Comment (lilin90) [reply en/upgrade-tidb-operator.md:85]: Confirmed. Just keep the capitalization.

**Pull request merged**